spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -422,13 +421,8 @@ private RewriteFiles(String fileSetID) {
     @Override
     public void commit(WriterCommitMessage[] messages) {
       FileRewriteCoordinator coordinator = FileRewriteCoordinator.get();
-
-      Set<DataFile> newDataFiles = Sets.newHashSetWithExpectedSize(messages.length);
-      for (DataFile file : files(messages)) {
-        newDataFiles.add(file);
-      }
-
-      coordinator.stageRewrite(table, fileSetID, Collections.unmodifiableSet(newDataFiles));
+      Set<DataFile> newDataFiles = ImmutableSet.copyOf(files(messages));
```

Comment:
just wondering if using an `ImmutableSet` here has any problems in terms of serialization in spark potentially?

Reply:
Maybe good to add a test. Examples are here: https://github.com/apache/iceberg/pull/5437/files#diff-260d07a95eec0ca51526bc88fce0a8a86b0df2ec54fe3d705b0bfe9db9ae2db1R104-R124